### ISSUE TITLE
scrapbook with about 3100 notes giving very poor performance

### ISSUE THREAD

**Issue by grotesque** [danny0838/firefox-scrapbook#54]
"Update cache for full text search" and "Calculation of size" takes more than 15 min.
The storage I'm using is Samsung 850pro SSD.

Same issue is not seen with scrapbook plus.

**Comment by grotesque** [comment]
Update cache gets stuck in : "Removing...  urn:scrapbook:item20140526034958#index"

**Comment by danny0838** [comment]
Which version are you using? Does it present in our latest version [1.12.0](https://github.com/danny0838/firefox-scrapbook/wiki/History)?

**Comment by grotesque** [comment]
Yes.

Other symptoms :
1) the issue was seen only during the first "Update cache for full text search" and "Calculation of size"
executing. After I let it finish one time, later execution of the same command have acceptable performance.

2) Presently, while "Calculation of size" is executing, we see "Now Calculating" as (x/3127) 
but after "Calculation of size" finished we see Total Size: 684.19 MB (3103 Items)

3103 != 3127

I believe this could be a legacy issue with scrapbook, not due to your changes.
I also think this issue is related to this specific scrapbook because other larger scrapbook are not giving any problems.

Take this at very low priority because the issue occurs only once.

**Comment by danny0838** [comment]
Scrapbook X 1.12.0a50 has fixed an issue that update cache makes Firefox unresponsive if the .rdf is large. It's not expected to happen in 1.12.0. If it really does, please provide more detail since it could be some unaware new issue.

It would be a normal behavior if the first update cache is slower in the first run since items not updated since the latest cache are skipped.

As for the "Calculate" dialog, the count shown during calculating is the total entries in the .rdf file, while  the count after finished is the number of directories in the data folder. It is somehow misleading, though, and we'll adjust the message to make it more clear in the next release.

**Comment by danny0838** [comment]
The "Calculate" issue is fixed in d53f6c401c16f3b3b39cbada0415f88d7b252988.

**Comment by xtraeme-xt** [comment]
I recently migrated over from the original Scrapbook extension by Gomita (love all the improvements!). The scrapbook I use has 10,255 objects and the cache.rdf is usually about ~330 MBs which is I realize quite large. Deleting and attempting to recreate the cache file with Scrapbook X takes hours to get close to 100% and when it's almost done I hit the same problem @grotesque reported in https://github.com/danny0838/firefox-scrapbook/issues/54#issuecomment-99174236. Reverting back to Gomita's scrapbook the file generates within a couple of minutes no problem. I tried diffing between the Scrapbook X 1.12.30 xpi after uncompressing the contents and compared it to Scrapbook 1.5.13.1-signed, but I can see there are massive number of changes in [fulltext.js](https://github.com/danny0838/firefox-scrapbook/blob/master/chrome/content/scrapbook/fulltext.js), [output.js](https://github.com/danny0838/firefox-scrapbook/blob/master/chrome/content/scrapbook/output.js), and [scrapbook.js](https://github.com/danny0838/firefox-scrapbook/blob/master/chrome/content/scrapbook/scrapbook.js) so I'm not sure I'd be much help trying to debug and narrow down on the problem. An old copy of the scrapbook is available online if you'd like to see the problem for yourself: https://mega.nz/#!MdMABIaA!poaKbM1CoSiH-4ikqNJ5ldKFX5QvLhoRasftgWmTdlw  Appreciate the help and all your work with the improvements!

**Comment by danny0838** [comment]
Close issue due to no further response for a year. It's welcome to refile it if similar problem still persists.
